src/containers/AppContainer.js

```diff
@@ -38,19 +38,57 @@ class AppContainer extends Component {
     var savedVfrCharts = getSavedCharts().map(function(chart) {
       return new VFRChart(chart);
     });
-
     this.setState({
       savedVfrCharts: savedVfrCharts,
       vfrChartsToShow: savedVfrCharts
     });
   }
+  //handle just like menu press so update both savedVfrCharts and vfrChartsToShow
+  handleFavPress(chartId: number){
+    let savedVfrCharts = [];
+    let vfrChartsToShow = [];
+    let vfrChart;
+
+    for(let i = 0; i < this.state.savedVfrCharts.length; i++){
+      savedVfrCharts[i] = this.state.savedVfrCharts[i];
+      if(savedVfrCharts[i].uniqueId == chartId){
+        savedVfrCharts[i].isFavorited = !savedVfrCharts[i].isFavorited;
+      }
+    }
+
+    switch(this.state.route){
```

Comment:
I know I kind of started this way of doing it, but maybe we should just call `setState({savedVfrCharts: savedVfrCharts})` in this method and `handleMenuPress` and then do the logic of figuring out what `vfrChartsToShow` should be in the `getCurrentSceneForRoute` method. 

And now that I’m thinking about it (I’m guilty of starting this), we really shouldn’t be storing `vfrChartsToShow` in the state at all since we can calculate that from the current `route` and `savedVfrCharts`. The React idea is to not store values that can be calculated in the state.

So this should be refactored to toggle the `isFavorited` ivar of the model, then set the state with the new list. Then `getCurrentSceneForRoute` could look like this:

```
getCurrentSceneForRoute() {
    switch (this.state.route.toLowerCase()) {
      case Scenes.HOME:
        return <VFRChartsList
                    onChartPress={(chartId) => this.handleFavPress(chartId)}
                    onViewPress={() => this.handleViewPress()}
                    vfrChartsToShow={this.state.savedVfrCharts}
                />;
      case Scenes.FAVORITES:
        return <VFRChartsList
                    onChartPress={(chartId) => this.handleFavPress(chartId)}
                    onViewPress={() => this.handleViewPress()}
                    vfrChartsToShow={ this.state.savedVfrCharts.filter((chart) => { return chart.isFavorited; }) }
                />;
      case Scenes.SETTINGS:
        return <Settings />;
      default:
        console.log("Unkown route: ", this.state.route);
        return <VFRChartsList onChartPress={(chartId) => this.handleFavPress(chartId)} vfrChartsToShow={this.state.savedVfrCharts}/>;
    }
  }
```

or something...

And `handleMenuPress` could be:
```
handleMenuPress(route: string) {
    this.setState({
      route: route,
      openMenu: false,
    });
  }
```

So now all of our filtering logic would be in `getCurrentSceneForRoute`